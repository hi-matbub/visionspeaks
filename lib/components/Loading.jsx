import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Spinner from 'react-bootstrap/Spinner';

const Loading = ({ data }) => {
  const [loading, setLoader] = useState(true);

  useEffect(() => {
    if (data) setLoader(false);
  }, [data]);

  return (
    <main className="full_screen loading">
      {loading ? <Spinner animation="grow" variant="warning" /> : null}
    </main>
  );
};

Loading.propTypes = {
  // eslint-disable-next-line react/require-default-props
  data: PropTypes.shape({}),
};

export default Loading;
