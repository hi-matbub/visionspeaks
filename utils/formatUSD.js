/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  // for inconsistent types
  const string = JSON.stringify(parseFloat(number));
  const { length } = string || number;
  const str = `${string.substring(0, length - 2)}.${string.substring(
    length - 2
  )}`;
  return parseFloat(str).toFixed(2);
};

export { formatUSD };
