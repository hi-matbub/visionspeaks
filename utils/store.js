exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.svg',
  hero: '/hero.svg',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
