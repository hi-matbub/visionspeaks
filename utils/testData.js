// used for testing purposes

exports.testData = {
  store: {
    name: 'Dog-Gone Goods',
    logo: '/logo.svg',
    hero: '/hero.svg',
    url: '',
    description: 'A dog treat shop',
    about: 'Everyone loves a happy dog!',
    contact: 'youremail@here.com',
    social: { facebook: '', twitter: '', instagram: '' },
  },
  products: [
    {
      id: 'prod_IRbpyL6fgse3n',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151601,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Large Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153202,
    },
    {
      id: 'prod_IRbouvicOUom16',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151551,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153221,
    },
  ],
};
