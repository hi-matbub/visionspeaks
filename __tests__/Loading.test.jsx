import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Loading from '../lib/components/Loading';
import { testData } from '../utils/testData';

it('Loading components renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Loading data={testData} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Loading component renders unchanged', () => {
  const tree = renderer.create(<Loading data={testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
