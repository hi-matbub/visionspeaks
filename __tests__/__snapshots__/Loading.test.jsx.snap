// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders unchanged 1`] = `
<main
  className="full_screen loading"
>
  <div
    className="spinner-grow text-warning"
  />
</main>
`;
